docs/source/coding_conventions.rst

```diff
@@ -0,0 +1,119 @@
+Coding Conventions
+==================
+
+In order to increase readability across Tensor Comprehensions written by
+multiple authors and to reduce the amount of surprising behavior, the
+following conventions should be adopted when writing TC. Generally in TC one
+should increment nesting by 4 whitespaces at each level and align tensor names
+and indices where appropriate to make memory access patterns emerge. Since
+these two goals can easily be conflicting, use your best judgement to tradeoff
+between the two goals. Such examples are provided below.
+
+Use indices named after parameters
+----------------------------------
+
+Use upper-case names for parameters and input/output tensors.
```

Comment:
`output`, `input` and many other tensor names in the examples do not use upper case.  I suppose we don't want to  uppercase the entire name, just to capital case it.